planner:  
    name: "lazy_prm_star"                          # Planner [rrt_star, rrt_sharp, lazy_prm_star, prm_star]
    elevation_layer: "elevation"                   # Layer name which has elevation information.
    plan_time: 10.0                                 # [s]   Planning duration.
    n_threads: 1                                   # Number of threads for ROS spinner.
    replan_freq: 0.5                               # [Hz]    Frequency of replanning.
    simplify_solution: false                       # Simplify the solution path before publishing.
    snap_goal_to_map: true                         # Get z, roll, pitch from elevation map.

sampler:        
    max_pitch_pert: 10                              # [degree]  Maximum pitch pertubation when sampling.
    max_roll_pert: 3.33                             # [degree]  Maximum roll pertubation when sampling.
    sample_from_distribution: true                  # Whether to use our custom sampling distribution or sample uniformly in the map.
    use_inverse_vertex_density: true                # Only works with "lazy_prm_star_min_update". Sampling biased towards regions with low-density of graph vertices.
    use_max_prob_unknown_samples: true              # Limit drawing samples from unknown map region to certain probability.
    max_prob_unknown_samples: 0.1                   # The maximum probability of drawing samples from the unknown region.
    reach_min_phi: false                            # Enforce the sampler to reach the min value of phi
    consider_theta: true                        # Consider y-axis angle for more safety

validator:
    max_motion_slope: 20.0                          # [degree] Maximum slope angle allowed for a motion
    max_motion_length: 1.0                         # [m] Maximum length allowed for a motion
    min_motion_length: 0.5                          # [m] Minimum length allowed for a motion
    max_motion_yaw_diff: 20.0                       # [degree] Maximum yaw difference allowed between two states in motion
    max_motion_slope_yaw_diff_free: 2.0             # [degree] Maximum motion slope angle to neglect motion yaw difference threshold
    max_state_phi: 20.0                             # [degree] Maximum x-axis angle for each state
    max_state_theta: 20.0                           # [degree] Maximum y-axis angle for each state

robot:      
    base_frame: "base_link"                              # Frame ID which is used to get robot pose.
    torso:
        length: 1.10                                # [m] Length.
        width: 1.0                                  # [m] Width.
        height: 0.5                                 # [m] Height.
        offset:                                     # [m] Torso center offset relative to base_frame.
            x: 0.0
            y: 0.0
            z: 0.04
    feet:
        offset:                                     # [m] Reachability box offset relative to base_frame.
            x: 0.51                                # Sign of value does not matter.
            y: 0.2                                # Sign of value does not matter.
            z: -0.475
        reach:                                      # [m] Reachability of foot box size.
            x: 0.2
            y: 0.2
            z: 0.2

verbose: true
